apps/consoledot-rhosak/src/routes/data-plane/routes/TopicCreateRoute.tsx

```diff
@@ -0,0 +1,110 @@
+import type { VoidFunctionComponent } from "react";
+import { useCallback } from "react";
+import { CreateTopic } from "ui";
+import type { Topic } from "ui-models/src/models/topic";
+import { useHistory } from "react-router-dom";
+import { addNotification } from "@redhat-cloud-services/frontend-components-notifications";
+import { useDispatch } from "react-redux";
+import { useCreateTopicMutation, useTopics } from "consoledot-api";
+import { useDataPlaneGate } from "../useDataPlaneGate";
+import type { DataPlaneNavigationProps } from "../routesConsts";
+import {
+  developerDefaults,
+  standardDefaults,
+} from "consoledot-api/src/transformers/topicTransformer";
+
+export const TopicCreateRoute: VoidFunctionComponent<
+  DataPlaneNavigationProps
+> = ({ instancesHref, instanceTopicsHref }) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const history = useHistory();
+  const { instance } = useDataPlaneGate();
+  const createTopic = useCreateTopicMutation();
+  const dispatch = useDispatch();
+  const availabiltyZone = instance.az;
+  const { data: topics } = useTopics({
+    id: instance.id,
+    adminUrl: instance.adminUrl,
+    plan: instance.plan,
+  });
+
+  const checkTopicName = (topicName: string) => {
```

Comment:
```suggestion
  const checkTopicName: CreateTopicProps['checkTopicName'] = (topicName) => {
```

This is longer to type, but binds the function type to where it's used. It works as a hint to the reader about where the function is going to be used.

Also, the other callbacks are wrapped in `useCallback`. This should be as well.